core/modules/file/file.entity.inc

```diff
@@ -141,6 +141,18 @@ class File extends Entity {
     );
   }
 
+  /**
+   * Overrides Entity::createAccess().
+   */
+  public static function createAccess($bundle = NULL, $account = NULL) {
+    // If no user object is supplied, the access check is for the current user.
+    if (empty($account)) {
+      $account = $GLOBALS['user'];
```

Comment:
@indigoxela is this check here really needed? `$account` is only ever used in `user_access()` below, and `user_access()` already checks for this.

Reply:
I tried to stay consistent with the other createAccess implementation. They all do it that way (node, user, taxonomy, comment).